components/ILIAS/Forum/classes/Moderation/class.ForumModeratorsTable.php

```diff
@@ -133,80 +156,29 @@ public function getTotalRowCount(
                 ?array $filter_data,
                 ?array $additional_parameters
             ): ?int {
-                return count($this->getRecords());
+                $this->initRecords();
+                return count((array) $this->records);
             }
 
-            protected function getRecords(Data\Range $range = null): array
+            private function getRecords(Data\Range $range, Data\Order $order): array
             {
-                $records = [];
-                $i = 0;
-                $entries = $this->forum_moderators->getCurrentModerators();
-                $num = count($entries);
-                foreach ($entries as $usr_id) {
-                    /** @var ilObjUser $user */
-                    $user = ilObjectFactory::getInstanceByObjId($usr_id, false);
-                    if (!($user instanceof ilObjUser)) {
-                        $this->oForumModerators->detachModeratorRole($usr_id);
-                        continue;
-                    }
+                $this->initRecords();
+                $records = $this->records;
 
-                    if ($num > 1) {
-                        $records[$i]['usr_id'] = $user->getId();
-                    } else {
-                        $records[$i]['usr_id'] = 0;
-                    }
-                    $records[$i]['login'] = $user->getLogin();
-                    $records[$i]['firstname'] = $user->getFirstname();
-                    $records[$i]['lastname'] = $user->getLastname();
-                    ++$i;
-                }
-
-                if ($range) {
-                    $records = $this->limitRecords($records, $range);
-                }
-
-                return $records;
-            }
-
-            protected function orderRecords(array $records, Data\Order $order): array
-            {
-                [$aspect, $direction] = $order->join("", function ($i, $k, $v) {
-                    return [$k, $v];
+                [$order_field, $order_direction] = $order->join([], fn ($ret, $key, $value) => [$key, $value]);
+                usort($records, static function (array $left, array $right) use ($order_field): int {
+                    return $left[$order_field] <=> $right[$order_field];
```

Comment:
I had a look at the legacy implementation.

Maybe it is better to use `\ilArrayUtil::sortArray` (see usage in `\ilTable2GUI::getHTML`) if you want to sort the array or `\ilStr::strCmp` if you want to compare strings (not numbers) instead of using the `<=>` operator.

Reply:
Ok. I added a workaround until we have a decision about how to handle the problem in future. For now, I decided to use ilArrayUtil::stableSortArray().